packages/svelte/src/internal/client/proxy/readonly.js

```diff
@@ -26,25 +24,26 @@ export function readonly(value) {
 		typeof value === 'object' &&
 		value != null &&
 		!is_frozen(value) &&
+		STATE_SYMBOL in value && // TODO handle Map and Set as well
 		!(READONLY_SYMBOL in value)
 	) {
-		const prototype = get_prototype_of(value);
-
-		// TODO handle Map and Set as well
-		if (prototype === object_prototype || prototype === array_prototype) {
-			const proxy = new Proxy(value, handler);
-			define_property(value, READONLY_SYMBOL, { value: proxy, writable: false });
-
-			return proxy;
-		}
+		const proxy = new Proxy(value, handler);
+		define_property(value, READONLY_SYMBOL, { value: proxy, writable: false });
+		return proxy;
 	}
 
 	return value;
 }
 
-/** @returns {never} */
-const readonly_error = () => {
-	throw new Error(`Props cannot be mutated, unless used with \`bind:\``);
+/**
+ * @param {any}	_
+ * @param {string} prop
+ * @returns {never}
+ */
+const readonly_error = (_, prop) => {
+	throw new Error(
+		`Props cannot be mutated, unless used with \`bind:\`. Use \`bind:prop-in-question={..}\` to make \`${prop}\` settable. Fallback values can never be mutated.`
```

Comment:
This should have a short and long form depending on the value of `DEV` like several other runtime errors, right?

Reply:
This only shows up during DEV, so doesn't need that

Reply:
Gotcha, thanks!